packages/core/ios/LottieReactNative/ContainerView.swift

```diff
@@ -190,43 +197,58 @@ class ContainerView: RCTView {
         applyProperties()
     }
 
+    // There is no Nullable CGFloat in Objective-C, so this function uses a Nullable NSNumber and converts it later
+    @objc(playFromFrame:toFrame:)
+    func objcCompatiblePlay(fromFrame: NSNumber? = nil, toFrame: AnimationFrameTime) {
+        let convertedFromFrame = fromFrame != nil ? CGFloat(truncating: fromFrame!) : nil;
+        play(fromFrame: convertedFromFrame, toFrame: toFrame);
+    }
+    
     func play(fromFrame: AnimationFrameTime? = nil, toFrame: AnimationFrameTime) {
         let callback: LottieCompletionBlock = { animationFinished in
             if let onFinish = self.onAnimationFinish {
                 onFinish(["isCancelled": !animationFinished])
             }
+            self.delegate?.onAnimationFinish(isCancelled: !animationFinished);
         }
 
         animationView?.backgroundBehavior = .pauseAndRestore
         animationView?.play(fromFrame: fromFrame, toFrame: toFrame, loopMode: self.loop, completion: callback);
     }
 
-    func play() {
+    @objc func play() {
         let callback: LottieCompletionBlock = { animationFinished in
             if let onFinish = self.onAnimationFinish {
                 onFinish(["isCancelled": !animationFinished])
             }
+            self.delegate?.onAnimationFinish(isCancelled: !animationFinished);
         }
 
         animationView?.backgroundBehavior = .pauseAndRestore
         animationView?.play(completion: callback)
     }
 
-    func reset() {
+    @objc func reset() {
         animationView?.currentProgress = 0;
         animationView?.pause()
     }
 
-    func pause() {
+    @objc func pause() {
         animationView?.pause()
     }
 
-    func resume() {
+    @objc func resume() {
         play()
     }
 
+    // reset the animationView frame whenever the view's frame changes
+    override var frame: CGRect {
```

Comment:
By the time we set the animationView's frame, React might not have calculated the view's frame so we need to change the animationView's frame when the view's frame changes.